libraries/GCS_MAVLink/MissionItemProtocol_Waypoints.cpp

```diff
@@ -52,6 +52,11 @@ bool MissionItemProtocol_Waypoints::clear_all_items()
 
 MAV_MISSION_RESULT MissionItemProtocol_Waypoints::complete(const GCS_MAVLINK &_link)
 {
+    if (!mission.check_do_jump_tags()) {
+        _link.send_text(MAV_SEVERITY_WARNING, "Mission upload failed due to DO_JUMP_TAG verification");
```

Comment:
Wait, haven't we stored the thing at this point?!

Reply:
Yes, so it's equivalent to sending an out of range DO_Jump as your last item.